#156882159 User can create or start a new game

amarachukwu-agbo opened the pull request:
#### What does this PR do?
Lets a user start a game after clicking a button displayed on a modal. It also creates an endpoint to save the game if the user is authenticated with a token.

#### Description of Task to be completed?
N/A

#### How should this be manually tested?
##### Front-end
When a user clicks on the button Play game with friends or Play game with strangers, he is redirected to the game page which displays a modal prompting the user to click a button to start the game or abandon the game. When the start game button is clicked, the game starts.

##### Back-end
Open Postman and sign up with the url *127.0.0.1:4000/api/auth/signup*. Get the token issued on sign up and include it in a header (card-game-token). Make POST request to *127.0.0.1:4000/api/games//start* and you should receive status code of *201* and *message: Success, Game Saved*.

#### Any background context you want to provide?
Sign up endpoint requires fields: username, email, password, imageUr.
Start game endpoint requires fields: players(Array), winner(String), gameStarter(String) and roundsPlayed(Number)

N/A

#### What are the relevant pivotal tracker stories?
#156882159

#### Screenshots (if appropriate)
N/A

#### Questions: N/A

amarachukwu-agbo committed:
chore(travis): set up travis ci integration
- enable continuous integration on the repository
[Delivers #156882150]

amarachukwu-agbo committed:
chore(travis): set up travis ci integration
- add travis badge to readme
[Delivers #156882150]

amarachukwu-agbo committed:
chore(travis): set up travis ci integration
- add travis badge to readme
[Delivers #156882150]

amarachukwu-agbo committed:
chore(travis): set up travis ci integration
- adhere to PR standards
[Delivers #156882150]

amarachukwu-agbo committed:
chore(travis): set up travis ci integration
- add travis badge to readme file
[Delivers #156882150]

amarachukwu-agbo committed:
chore(travis): set up travis ci integration
- add database service to travis.yml
- add script to get coverage report
[Delivers #156882150]

amarachukwu-agbo committed:
merge staging

amarachukwu-agbo committed:
feat(game): user can create/start game
- ensure user is authenticated
- display modal to let user start game
- save game when it ends

[Delivers #156882159]

kenoseni reviewed (commented):
Good job Amara for implementing this feature, but you need to update this branch with Sunday's new pull request so that the avatar functionality can be restored and the game can start without crashing the server.

amarachukwu-agbo committed:
chore(start a game): user should be able to start a new game

- display modal to inform user he is about to start a game
- create endpoint to save game when it ends
- test created endpoint

[Delivers #156882159]

Orlayhemmy reviewed (commented):
Is there any reason why you increased the timeout for the test script in the package.json
Endeavour to remove console.log from your code
In "test/game/startgame.test.js" on line 26, does that not stop guest from starting a game
